enclave_mgmt/models.py

```diff
@@ -0,0 +1,223 @@
+from pydantic import BaseModel, Extra, validator
+from typing import List, Union
+from uuid import UUID
+from datetime import datetime
+from typing import Literal
+from math import isnan
+
+
+class Demographic(BaseModel):
+    user_uuid: UUID
+    birth_date: str
+    sex: Literal['male', 'female']
+    american_indian_or_alaska_native: Literal['true', 'false']
+    asian: Literal['true', 'false']
+    black_or_african_american: Literal['true', 'false']
+    native_hawaiian_or_other_pacific_islander: Literal['true', 'false']
+    white: Literal['true', 'false']
+    demographic_race_two_or_more_races: Literal['true', 'false']
+    hispanic_or_latino_ethnicity: Literal['true', 'false']
+
+    class Config:
+        extra = Extra.forbid
+
+    @validator('birth_date')
+    def birthdate_format(cls, v):
+        datetime.strptime(v, "%Y-%m-%d")
+        return v
+
+
+class Assessment(BaseModel):
+    id: int
+    name: str
+
+    class Config:
+        extra = Extra.forbid
+
+
+class Course(BaseModel):
+    id: int
+    name: str
+
+    class Config:
+        extra = Extra.forbid
+
+
+class Enrollment(BaseModel):
+    user_uuid: UUID
+    course_id: int
+    role: Literal['student', 'teacher']
+
+    class Config:
+        extra = Extra.forbid
+
+
+class Grade(BaseModel):
+    assessment_id: int
+    user_uuid: UUID
+    course_id: int
+    grade_percentage: float
+    time_submitted: int
+
+    class Config:
+        extra = Extra.forbid
+
+    @validator('grade_percentage')
+    def grade_value(cls, v):
+        if isnan(v):
+            raise ValueError('Grade value is nan')
```

Comment:
Maybe it's worth adding some test coverage on these model validators? We could create a `test_models.py` or something for that.